search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/llm/ChatCompletionOutput.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.searchpipelines.questionanswering.generative.llm;
+
+/**
+ * Outputs from chat completion API.
+ */
+public interface ChatCompletionOutput {
+    String getAnswer();
```

Comment:
string as an output choice for this function is will make the scope of the function very limited. Why we are thinking of this in terms of an Abstract Answer class for a model output. Example : There can be multiple answer provided

Reply:
As per my understanding we cannot predict that there will always be 1 ans string. Hence I would recommend making this interface more robust for future usecase where some model can provide more than 1 string.

Reply:
If you have a model that returns multiple answers, how would the client know which one to pick?  We are talking about a specific operation of LLMs that give you a single answer.  If we need to handle a case where a model returns multiple answers for a different use case (it would not be RAG), then we can add a new method to the Output interface (and maybe even a new method to the Llm interface) to accommodate such a use case.  Here, we are not aiming for a single method/output that is generic enough to handle all future/unknown use cases.

Reply:
Client can always pick the top one, or if there are more processing needs to be done on those results with a complex object clients can do that.

Reply:
But today, we do not know the "shape" of those objects.  I think you are thinking of a `List` of `class Foo` here, but I think that's overkill knowing that this is the output of `chat completion` meaning we know it's a string.  For instance, for OpenAI, it's a `content` string inside a `Message`.

You may be thinking that since the search engine returns a list of hits (with scores), the (LLM) model can do the same.  I am pointing out that this is a ChatCompletion API where the functionality has more to do with generating an answer (synthesizing multiple contexts into one).

Reply:
See having string make the interface locked and in future we cannot add more attributes if needed. Hence the suggestion was to move to an Object, which gives flexibility for future.